browser/ui/views/brave_ads/ad_notification_popup.cc

```diff
@@ -347,6 +231,24 @@ void AdNotificationPopup::AnimationCanceled(const gfx::Animation* animation) {
   UpdateAnimation();
 }
 
+AdNotification AdNotificationPopup::GetAdNotification() const {
+  return ad_notification_;
+}
+
+void AdNotificationPopup::MovePopup(const gfx::Vector2d& movement) {
+  gfx::Rect bounds = CalculateBounds() + movement;
+  AdjustBoundsAndSnapToFitWorkAreaForWidget(GetWidget(), &bounds);
+}
+
+void AdNotificationPopup::ClosePopup() {
+  const std::string notification_id = ad_notification_.id();
+  DCHECK(!notification_id.empty());
+
+  AdNotificationPopupCollection::Erase(notification_id);
```

Comment:
moved `g_ad_notification_popups` cleanup here to avoid possible races on popup close.